Yannick451 - Added Random Forest Variable Importance Plot

yvanhuele opened the pull request:
Changes:
- Added row numbers to predictions in random forest development (so the test set can be determined from getPredictions(), similar to lasso).
- Added a getter function for the random forest variable importance list, with the option to specify the number of variables to include (includes all by default)
- Added a variable importance plot for random forest
- Documented both functions

mmastand commented on R/random-forest-development.R:
```diff
@@ -461,6 +486,35 @@ RandomForestDevelopment <- R6Class("RandomForestDevelopment",
+      return(private$variableImportanceList[1:top, ])
+    },
+    
+    plotVariableImportance = function(numTopVariables = NULL) {
```
I found an issue with the y-axis text if the feature names are long. See below:
![image](https://user-images.githubusercontent.com/15145347/29690824-c7541586-88e5-11e7-88f4-f5c017cfa404.png)

mmastand commented on R/random-forest-development.R:
```diff
@@ -461,6 +486,35 @@ RandomForestDevelopment <- R6Class("RandomForestDevelopment",
+    },
+    
+    plotVariableImportance = function(numTopVariables = NULL) {
+      # Get information for plot before chaning margins
```
changing

mmastand commented on R/random-forest-development.R:
```diff
@@ -461,6 +486,35 @@ RandomForestDevelopment <- R6Class("RandomForestDevelopment",
+      # Change margins in tryCatch so that margins are always reset even if
+      # plotting fails
+      tryCatch({
+        # indent plot so that variable names can be read
```
indent more?

mmastand commented on R/random-forest-development.R:
```diff
@@ -461,6 +486,35 @@ RandomForestDevelopment <- R6Class("RandomForestDevelopment",
+    },
+    
+    getVariableImportanceList = function(numTopVariables = NULL) {
+      top <- min(numTopVariables, nrow(private$variableImportanceList))
```
Never mind, there's the check.

yvanhuele commented:
New changes:
- abbreviated long labels: for labels with more than 23 characters, only the first and last 10 characters are displayed in the plot and the variable importance list is printed in the console
- adjusted margins to deal with longer labels (not perfect, but seems to work)
- added warning if `plotVariableImportance` or `getVariableImportanceList` are asked to display more variables than were used to train the random forest

Examples:

![abbreviated2](https://user-images.githubusercontent.com/8187218/29724620-4d0f5a5a-8986-11e7-9721-a45ee12030a8.png)

![abbreviated1](https://user-images.githubusercontent.com/8187218/29724614-447e62e6-8986-11e7-8db3-96762ab6fda1.png)
